### DISCUSSION TITLE
How to handle dynamic content with showError

### DISCUSSION THREAD

**Discussion by Luffyyy** [nuxt/nuxt#16919] [Questions] [answered]
So we have this example from the docs: 

```vue
<script setup>
const route = useRoute()
const { data } = await useFetch(`/api/movies/${route.params.slug}`)
if (!data.value) {
  throw createError({ statusCode: 404, statusMessage: 'Page Not Found' })
}
</script>
```

But this doesn't really cover what to do with the template. Do we do a v-if with the data? Because from my testing I've done, Nuxt doesn't seem to stop the page once there's an error and attempts to access the null data anyway causing an additional error. I think this part needs the template part too.

**Comment by danielroe** [comment, marked as answer]
This is now resolved in https://github.com/nuxt/framework/pull/7290.
